### PR TITLE
add resolver for acm certificates

### PR DESCRIPTION
**Pull request by rbayerl**
Partially addresses https://github.com/envato/stack_master/issues/92

@stevehodgkiss I'm not sure what you're looking for wrt EC2 KeyPairs or EIP allocation IDs.

EC2 key pairs are specified by key name in Cloudformation so I'm not certain how useful a resolver would be or what it would resolve.

EIP associations are done on the EIP resource (not the instance itself) so not sure what to resolve here either. Is the use case that given `x` allocation ID just return the public IP? What's the use case?

Happy to add more with clarification 😄

### CODE REVIEW

File: lib/stack_master/parameter_resolvers/acm_certificate.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    class AcmCertificate < Resolver
+      CertificateNotFound = Class.new(StandardError)
+
+      def initialize(config, stack_definition)
+        @config = config
+        @stack_definition = stack_definition
+      end
+
+      def resolve(value)
+        cert_arn = all_certs.map { |c| c.certificate_arn if c.domain_name == value }.compact.first
```
Comment (stevehodgkiss) [inline lib/stack_master/parameter_resolvers/acm_certificate.rb]: Maybe a extract method with this logic (`find_cert_arn_by_domain_name`?)

Comment (rbayerl) [reply lib/stack_master/parameter_resolvers/acm_certificate.rb]: :+1:

File: lib/stack_master/parameter_resolvers/acm_certificate.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      end
+
+      private
+
+      def all_certs
+        certs = []
+        next_token = nil
+        client = Aws::ACM::Client.new(region: @stack_definition.region)
+        loop do
+          resp = client.list_certificates(next_token: next_token)
```
Comment (patrobinson) [inline lib/stack_master/parameter_resolvers/acm_certificate.rb]: Should we filter for only `ISSUED` certificates?

Comment (rbayerl) [reply lib/stack_master/parameter_resolvers/acm_certificate.rb]: Good call. Done.

**Comment by patrobinson** [comment]
@rbayerl I'm OK with dropping support for 2.1

**Comment by rbayerl** [comment]
Might need to clear the cache on Travis.

**Comment by rbayerl** [comment]
Could be I was wrong and the test just fails intermittently. Seems you didn't hit this issue with your recent PR. I can drop the last commit if needed.

**Comment by patrobinson** [comment]
@rbayerl I'm still ok with it, please resolve the merge conflict and I'll merge

**Comment by rbayerl** [comment]
Dropped that commit anyway (unintentionally).

**Comment by rbayerl** [comment]
@patrobinson anything else needed here?

**Pull request merged**